src/Cake.NuGet.Tests/Stubs/FakeNuGetSettings.cs

```diff
@@ -0,0 +1,46 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using NuGet.Configuration;
+
+namespace Cake.NuGet.Tests.Stubs
+{
+    internal sealed class FakeNuGetSettings : ISettings
+    {
+        public event EventHandler SettingsChanged;
+
+        private IDictionary<string, FakeNuGetSettingSection> _settings = new Dictionary<string, FakeNuGetSettingSection>(StringComparer.OrdinalIgnoreCase);
+
+        public void AddOrUpdate(string sectionName, SettingItem item)
+        {
+            if (!_settings.TryGetValue(sectionName, out var section))
+            {
+                section = new FakeNuGetSettingSection(sectionName, null, Enumerable.Empty<SettingItem>());
+                _settings[sectionName] = section;
+            }
+
+            section.AddItem(item);
+
+            SettingsChanged?.Invoke(this, new EventArgs());
+        }
+
+        public IList<string> GetConfigFilePaths() => Array.Empty<string>();
+
+        public IList<string> GetConfigRoots() => Array.Empty<string>();
+
+        public SettingSection GetSection(string sectionName) =>
```

Comment:
Don't use expression-bodied methods. We're not using it anywhere else in the code base.